coderd/rbac/authz.go

```diff
@@ -617,94 +626,37 @@ type cachedCalls struct {
 //
 // Cacher is safe for multiple actors.
 func Cacher(authz Authorizer) Authorizer {
-	return &cachedCalls{authz: authz}
+	return &authCache{
+		authz: authz,
+		cache: tlru.New[string](tlru.ConstantCost[error], 4096),
+	}
 }
 
-func (c *cachedCalls) Authorize(ctx context.Context, subject Subject, action Action, object Object) error {
-	cache := cacheFromContext(ctx)
+func (c *authCache) Authorize(ctx context.Context, subject Subject, action Action, object Object) error {
+	var authorizeCacheKey strings.Builder
+	enc := json.NewEncoder(&authorizeCacheKey)
+	_ = enc.Encode(subject)
+	_ = enc.Encode(action)
+	_ = enc.Encode(object)
```

Comment:
This is unfortunate. The json payload is quite large, and that being a key would be even more verbose.

It would be better to "hash" the fields somehow to reduce this.

I intentionally made the file [`astvalue.go`](https://github.com/coder/coder/blob/6ed35b406a7a0a3ec877ed41d11da904d243ad09/coderd/rbac/astvalue.go#L11-L33) to avoid json.Marshalling this data when going to the rego evaluator.

----

We can use the role name for the hash and ignore the permissions. Whatever "hash" function we create will be faster than a json.Marshal I bet.